File: 0788-rotated-digits/0788-rotated-digits.js
/**
 * @param {number} n
 * @return {number}
 */
var rotatedDigits = function(n) {
  let ans = 0;
  
  while (n) ans += isGood(n--);
  
  return ans;
};

const isGood = (n) => {
  let good = false;
  
  while (n) {
    const x = n % 10;
    if (x === 3 || x === 4 || x === 7) return false;
    if (x === 2 || x === 5 || x === 6 || x === 9) good = true;
    
    n = Math.floor(n / 10);
  }
  
  return good;
}

File: 0788-rotated-digits/0788-rotated-digits.py
class Solution:
    def rotatedDigits(self, n: int) -> int:
        def isGood(n):
            good = False
            while n:
                x = n % 10
                if x == 3 or x == 4 or x == 7:
                    return False
                if x == 2 or x == 5 or x == 6 or x == 9:
                    good = True
                n //= 10
            return good
          
        
        ans = 0
        while n: 
            ans += isGood(n)
            n -= 1
        return ans